pykern/pkcli/ci.py

```diff
@@ -8,59 +8,50 @@
 :copyright: Copyright (c) 2022 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
-from pykern.pkdebug import pkdp, pkdlog
-from pykern import pkunit
-from pykern.pkcli import test
+from pykern import pkcli
+from pykern import pkio
 from pykern import pksetup
+from pykern import pkunit
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp, pkdlog
 import re
 
-
-_FILE_TYPE = re.compile(r".py$")
-_EXCLUDE_FILES = re.compile(
-    f".*(?:{pkunit.DATA_DIR_SUFFIX}|{pkunit.WORK_DIR_SUFFIX})/"
-    + f"|^\\w+/{pksetup.PACKAGE_DATA}/"
-    + r"|pkdebug[^/]*\.py$"
-    + r"|(?:^|/)\."
-    + r"|^run/"
-)
 _PRINT = re.compile(r"(?:\s|^)(?:pkdp|print)\(")
 _PRINT_OK = re.compile(r"^\s*#\s*(?:pkdp|print)\(")
 
 
+def check_eof_newline():
+    """Recursively check repo for files missing newline at end of file.
+
+    Files specified in _check_files() in exclude/include_files will be checked.
```

Comment:
While "by reference" documentation is easy to maintain, it's not useful. Please denormalize the description. It doesn't have to be literal, e.g. it might say something like ignore directories that match run, test data/work, package_data, etc. Try to imagine reading this as an outsider on pykern.rtfd.io.

Reply:
a9e80a0beda32d4163def682c4cffe18637e6076